src/Lykke.Service.LiquidityEngine/Migration/StorageMigrationService.cs

```diff
@@ -0,0 +1,122 @@
+
+            _log.InfoWithDetails("Read current storage version.", systemVersion);
+
+            foreach (MigrationOperationDescription operation in operations)
+            {
+                if (versionNumber == operation.ApplyToVersion)
+                {
+                    _log.InfoWithDetails("Applying migration operation.", operation);
+
+                    await operation.Operation.ApplyAsync();
+                    versionNumber = operation.UpdatedVersion;
+
+                    _log.InfoWithDetails("Migration operation executed", operation);
+                }
+            }
+
+            if (versionNumber != systemVersion.VersionNumber)
+            {
+                systemVersion.VersionNumber = versionNumber;
+
+                _log.InfoWithDetails("Updating storage version", systemVersion);
+
+                await _versionControlRepository.UpdateAsync(systemVersion);
+            }
+        }
+
+        private static IReadOnlyList<MigrationOperationDescription> ValidateOperations(IReadOnlyCollection<IMigrationOperation> migrationOperations)
+        {
+            if (migrationOperations.Count == 0)
+            {
+                return new MigrationOperationDescription[0];
+            }
+
+            IReadOnlyList<MigrationOperationDescription> operations = migrationOperations
+                .Select(o =>
+                {
+                    StorageMigrationOperationAttribute migrationAttribute =
+                        (StorageMigrationOperationAttribute)Attribute.GetCustomAttribute(o.GetType(),
+                            typeof(StorageMigrationOperationAttribute));
+
+                    if (migrationAttribute == null)
+                    {
+                        throw new InvalidOperationException("MigrationOperationAttribute is not found.");
+                    }
+
+                    return new MigrationOperationDescription
+                    {
+                        Name = migrationAttribute.Name,
+                        ApplyToVersion = migrationAttribute.ApplyToVersion,
+                        UpdatedVersion = migrationAttribute.UpdatedVersion,
+                        Operation = o
+                    };
+                })
+                .OrderBy(o => o.ApplyToVersion)
+                .ToArray();
+
+            operations.Aggregate((prev, next) =>
+            {
+                if (prev.ApplyToVersion == next.ApplyToVersion)
+                {
+                    throw new InvalidOperationException("Migration operations have same ApplyTo versions.");
+                }
+
+                if (prev.UpdatedVersion > next.ApplyToVersion)
+                {
+                    throw new InvalidOperationException("Migration operations have inconsistent updated versions.");
+                }
+
+                return next;
+            });
+
+            return operations;
+        }
+
+        private class MigrationOperationDescription
```

Comment:
Remove migration attribute and move properties to the interface. This class no needed the operation should describe is self.

Reply:
Removed attribute.